Basic music player

ambern115 opened the pull request:
It plays youtube links!
Its organized better?

ambern115 committed:
Created a bot package and renamed a couple things

- I think something wasn't totally right with my python path because all
of the local file imports were broken. The best fix I could find was to
move all the bot-related code into a package.
- I also renamed basic.py to main.py because I thought it fit better but
thats just my opinion.
- I don't know if I like the name of the package, change it if you think
something else is better.

ambern115 committed:
Added youtube player to bot

ambern115 commented:
Theres probably still some junk left over from testing stuff that I could still remove.

posneg commented:
I'll be looking into this PR tomorrow

posneg commented on bot/music_player/player.py:
```diff
@@ -0,0 +1,181 @@
+            data = data['entries'][0]
+
+        filename = data['url'] if stream else ytdl.prepare_filename(data)
+        #return cls(discord.FFmpegPCMAudio(filename, **ffmpeg_options), data=data)
```
Let's remove the extra line here

posneg reviewed (changes requested):
Most of the changes look good -- I think really the thing I'd like to see the most is logging, as that'll help debug any issues that arise later.

You can see some existing examples of logging in some of the code I've written, and you can use the bot's logger or you can create a music player specific log/handler.

posneg commented on bot/music_player/player_cog.py:
```diff
@@ -0,0 +1,121 @@
+        automatically add bot to a voice channel if its not in one already.""",
+        name="play_local"
+    )
+    async def play_local(self, ctx, song_name, filter=""):
```
If you want to use multiword song names without quotation marks, you can remove the filter (perhaps make the filter a command per-guild?) argument and use the following function definition instead.
```suggestion
    async def play_local(self, ctx, *, song_name):
```

posneg replied:
After discussion, this sort of change would be in a separate PR.  We'll leave that for later and accept as-is.

ambern115 committed:
Added logging to player, fixed some bugs

- Fixed bug with skip skipping two songs each time
- Fixed bug with disconnects when bot is in multiple servers
- Removed unused code

ambern115 committed:
Added bot to PlayerCog

ambern115 committed:
Moved init logs to log after self.bot is set

ambern115 committed:
Fixed bugs I introduced with logs

posneg reviewed (approved):
LGTM for now
